PROJECT_STRUCTURE.md

```diff
@@ -0,0 +1,86 @@
+![Remixer](https://cdn.rawgit.com/material-foundation/material-remixer/master/docs/assets/lockup_remixer_icon_horizontal_dark_small.svg)
+
+# Project structure
+
+Remixer is built as a gradle project with several submodules. If you are interested in contributing you should know the roles of each submodule to be able to put things in the right place.
+
+## Core and Annotations
+
+Remixer has a core library, `remixer_core`, where the main logic for Remixer is, here is where you can find
+- DataTypes supported by Remixer `com.google.android.libraries.remixer.DataType`
+- Serialization logic `package com.google.android.libraries.remixer.serialization`
+- All variable logic `com.google.android.libraries.remixer.Variable`, `com.google.android.libraries.remixer.ItemListVariable`, `com.google.android.libraries.remixer.RangeVariable`
+- The `com.google.android.libraries.remixer.Remixer` class which aggregates all variables and `package com.google.android.libraries.remixer.sync` which takes care of keeping values in sync across different contexts.
+  - `com.google.android.libraries.remixer.sync.LocalValueSyncing` is a non-persistent version of a `com.google.android.libraries.remixer.sync.SynchronizationMechanism`. Persistent versions are available in the `remixer_storage` submodule, those depend on android-specific classes which are unavailable on this project, because of the reasons explained in the next section.
+- Declaration of all annotations exposed by Remixer `package com.google.android.libraries.remixer.annotation` and the annotation binding classes.
+
+The annotation-processing library, `remixer_annotation`, contains code that should only be run by the Java Compiler (and should never be in the classpath for an APK), this code generates Remixer variables from annotations in `com.google.android.libraries.remixer.annotation`.
+
+
+### Constraints on dependencies
+
+**TL;DR: If you contribute to Remixer, you need to know you cannot depend on any android classes in the `remixer_core` submodule**
```

Comment:
this sentence is a little ambiguous. Do you mean:
"If you contribute to Remixer, you need to know that the `remixer_core` submodule cannot depend on any android classes"